src/main/proto/wfa/measurement/protocol/liquid_legions_sketch_aggregation_v2.proto

```diff
@@ -28,31 +29,42 @@ message LiquidLegionsSketchAggregationV2 {
     // The computation stage is unknown. This is never set intentionally.
     STAGE_UNKNOWN = 0;
 
-    // The computation is known and the worker should confirm all required local
+    // The worker is in the initialization phase.
+    // More specifically, each worker will create a new ElGamal key pair solely
+    // used in this computation, and report the public key to the Kingdom.
+    INITIALIZATION_PHASE = 1;
+
+    // The computation is initialized locally. The worker is waiting for the
+    // list of local requisitions and the public ElGamal key set used in this
```

Comment:
Why is the mill waiting for public ElGamal key set from kingdom? It already generated and stored its ElGamal key that it will use in the computation right?

Reply:
The mill/duchy only know its own generated ElGamal key. It also needs to know the other duchies's public keys (coming from the kingdom after all duchies report their key to the kingdom). All the keys together determine the order of the duchies in the MPC ring. Also these keys are used in the computations.

Reply:
I thought combined ElGamal key is only used once, by the EDP when encrypting the tuples in the sketch it sends to duchy?
After that each duchy needs only its private key to decrypt its part of ElGamal combined key.
Are these two statements true? If so then the only reason for each duchy to know the other duchies keys is to know the ordering? They are public keys so I guess it doesn't really matter that they are known widely, but I am trying to understand.

Reply:
The duchies need to add various noise registers (encrypted with the same combined public key used by the EDPs), and also need to rerandomize ciphertexts that are not decrypted, which also need the combined public key.

Yes, all those public keys are available to all parties that need them.

Reply:
Ah right that makes sense, thanks!